tests/naive.py

```diff
@@ -1759,6 +1837,75 @@ def _total_diagonal_ndists(tile_lower_diag, tile_upper_diag, tile_height, tile_w
     return total_ndists
 
 
+def merge_topk_PI(PA, PB, IA, IB):
+    if PA.ndim == 1:
+        for i in range(PA.shape[0]):
+            if PB[i] < PA[i] and IB[i] != IA[i]:
+                PA[i] = PB[i]
+                IA[i] = IB[i]
+        return
+
+    k = PA.shape[1]
+    for i in range(PA.shape[0]):
+        _, _, overlap_idx_B = np.intersect1d(IA[i], IB[i], return_indices=True)
+        PB[i, overlap_idx_B] = np.inf
+        IB[i, overlap_idx_B] = -1
+
+    profile = np.column_stack((PA, PB))
+    indices = np.column_stack((IA, IB))
+    IDX = np.argsort(profile, axis=1, kind="mergesort")
+    profile[:, :] = np.take_along_axis(profile, IDX, axis=1)
+    indices[:, :] = np.take_along_axis(indices, IDX, axis=1)
+
+    PA[:, :] = profile[:, :k]
+    IA[:, :] = indices[:, :k]
+
+
+def merge_topk_ρI(ρA, ρB, IA, IB):
+    # this is to merge two pearson profiles `ρA` and `ρB`, where each is a 2D array
```

Comment:
fix comment: break it down to two parts: 1D array inputs and 2D array inputs (see performant version)